webapp/src/components/systemConsole/surveyResults/surveyRowActionButton.tsx

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import client from 'client/client';
+import React, {useCallback, useState} from 'react';
+
+import Button from 'components/common/button/button';
+import {ConfirmationModal} from 'components/systemConsole/surveyResults/confirmationModal';
+
+import type {SurveyResult} from 'types/plugin';
+
+type Props = {
+    surveyResult: SurveyResult;
+    onStopSurvey: (surveyID: string) => void;
+}
+
+export default function SurveyRowActionButton({surveyResult, onStopSurvey}: Props) {
+    const [isDownloadingReport, setIsDownloadingReport] = useState<boolean>(false);
+    const [showConfirmationModal, setShowConfirmationModal] = useState<boolean>(false);
+
+    const handleDownloadSurveyReport = useCallback(async (surveyID: string) => {
+        setIsDownloadingReport(true);
+        await client.downloadSurveyReport(surveyID);
+        setIsDownloadingReport(false);
+    }, []);
+
+    const handleShowConfirmationDialog = useCallback(() => {
+        setShowConfirmationModal(true);
+    }, []);
+
+    const handleHideConfirmationDialog = useCallback(() => {
+        setShowConfirmationModal(false);
+    }, []);
+
+    const handleStopSurvey = useCallback(() => {
+        onStopSurvey(surveyResult.id);
+    }, [onStopSurvey, surveyResult.id]);
+
+    if (surveyResult.status === 'ended') {
+        return (
+            <div
+                key={surveyResult.id}
+                className='surveyResultActions'
+            >
+                <Button
+                    iconClass={isDownloadingReport ? 'icon-refresh' : 'icon-download-outline'}
+                    text='Export responses'
+                    onClick={() => handleDownloadSurveyReport(surveyResult.id)}
```

Comment:
Now that this is a separate component, we can put this (and the one in line 68) as a `useCallback`, and we don't pass a new function on every render.

Reply:
Done.